txscript/sign.go

```diff
@@ -114,6 +133,39 @@ func TaprootWitnessSignature(tx *wire.MsgTx, sigHashes *TxSigHashes, idx int,
 	return wire.TxWitness{sig}, nil
 }
 
+// RawTxInTapscriptSignature computes a raw schnorr signature for a signature
+// generated from a tapscript leaf. This differs from the
+// RawTxInTaprootSignature which is used to generate signatures for top-level
+// taproot key spends.
+//
+// TODO(roasbeef): actually add code-sep to interface? not really used
+// anywhere....
+func RawTxInTapscriptSignature(tx *wire.MsgTx, sigHashes *TxSigHashes, idx int,
+	amt int64, pkScript []byte, tapLeaf TapLeaf, hashType SigHashType,
+	privKey *btcec.PrivateKey) ([]byte, error) {
+
+	// First, we'll start by compute the top-level taproot sighash.
+	tapLeafHash := tapLeaf.TapHash()
+	sigHash, err := calcTaprootSignatureHashRaw(
+		sigHashes, hashType, tx, idx,
+		NewCannedPrevOutputFetcher(pkScript, amt),
+		WithBaseTapscriptVersion(0, tapLeafHash[:]),
```

Comment:
This default value of `0` here is incorrect for the normal script spend. I needed to set it to `math.MaxUint32` in my case. See https://github.com/bitcoin/bips/blob/master/bip-0342.mediawiki#common-signature-message-extension, not sure where this would ever be something other than `0xffffffff`? I don't think I understand what "the opcode position of the last executed" means in this context. Is this in case we have multiple OP_CHECKSIG in the same script, we'd need to keep track of them and increment this value accordingly?

In any case, I think we should probably allow this to be set as a parameter here.

Reply:
Fixed this by having the `defaultTaprootSighashOptions` function use the max int as the default arg there, thanks for catching this!